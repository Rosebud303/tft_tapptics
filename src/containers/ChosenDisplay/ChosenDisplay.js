import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import { connect } from 'react-redux';
import Honeycomb from '../../components/Honeycomb/Honeycomb';
import HoneycombDrag from '../../components/HoneycombDrag/HoneycombDrag';
import Stats from '../Stats/Stats';

export class ChosenDisplay extends Component {
  constructor() {
    super();
    this.state = {
      currentTeam: [
        {
          name: 'vayne',
          health: 400,
          dmg: 260,
          img: 'https://i.imgur.com/3Fj3uOv.png'
        }
      ]
    }
  }
  
  render() {
    const displayChamps = this.props.champions.map( (champion, idx) => {
      return <HoneycombDrag key={champion.attributes.name}
                            id={champion.id}
                            idx={idx}
                            name={champion.attributes.name}
                            image={champion.attributes.champion_thumbnail} />
    });

    const displayRoster = this.props.roster.map( (rost, idx) => {
      if(rost === {}){
        return <Honeycomb key={idx} idx={idx} rost={rost}/>
      } else {
        return <Honeycomb key={idx} idx={idx} rost={rost}/>
      }
    })

    return (
      <div className='chosen-display'>
        <section className="current-roster">
          <section className="Honeycomb-section">
              {displayRoster}
          </section>
          <section>
              {displayChamps}
          </section>
          <article className="team-stats">
            <Stats currentTeam={this.state.currentTeam} />
          </article>
          <article className="team-synergies">
            <h3>Current Synergies:</h3>
            <h4>YORDLE</h4>
            <h4>SORCERER</h4>
          </article>
        </section>
      </div>
    )
  }
}

ChosenDisplay.propTypes = {
  champions: PropTypes.array
}

export const mapStateToProps = state => ({
  champions: state.champions,
  roster: state.setRoster
});

export default connect(mapStateToProps)(ChosenDisplay);