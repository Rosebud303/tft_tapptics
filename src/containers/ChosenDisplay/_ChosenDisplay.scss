.chosen-display {
  background-image: linear-gradient($fade1, $fade2), url('../../assets/images/summonersrift.jpg');
  height: 100vh;

  .choose-level {
    margin: auto;
    text-align: center;

    button {
      background-color: $btn-bg;
      border: 1px solid $btn-border;
      border-radius: 3px;
      color: darken($btn-border, 10%);
      display: block;
      font-size: .5rem;
      font-weight: bolder;
      height: 2vh;
      width: 2rem;

      &:hover {
        background-color: $static-accent;
        border: 1px solid $hover-accent;
        cursor: pointer;
      }

      &:focus {
        outline: $focus;
      }
    }

    .add-level {
      margin: .5rem 0;
    }

    h2 {
      display: inline;
      font-size: 2rem;
      margin: 0 .5rem;
      text-align: center;
    }

    .level-btns {
      display: inline-block;
    }
  }

  .current-roster {
    display: grid;
    grid-template-areas: 'honey1 stats synergy'
                          'honey2 stats synergy';
    grid-template-columns: 60% 20% 20%;
    grid-template-rows: 80% 20%;
    height: 60vh;

    article {
      text-align: center;
    }
  }

  .Honeycomb-section {
    display: flex;
    flex-wrap: wrap;
    grid-area: honey1;
  }

  .team-stats {
    background-color: $overlay;
    grid-area: stats;
    height: 80vh;
    padding: .5rem;
    width: 75%;
    overflow: scroll;

    img {
      display: inline-block;
    }
  }

  .team-synergies {
    background-color: $overlay;
    grid-area: synergy;
    height: 80vh;
    padding: .5rem;
    width: 75%;

    h3, h4 {
      margin: 0 auto;
    }
  }
}

.honey-2 {
  margin: 7.5rem 0;
  grid-area: honey2;
  margin-left: 1rem;
}