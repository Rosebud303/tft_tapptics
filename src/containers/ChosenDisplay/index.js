import React, { Component } from 'react'
import ChosenCharacter from '../../components/ChosenCharacters'

export default class ChosenDisplay extends Component {
  render() {
    return (
      <div className='chosen-display'>
        <h2>Current level:1</h2>
        <ChosenCharacter />
      </div>
    )
  }
}
