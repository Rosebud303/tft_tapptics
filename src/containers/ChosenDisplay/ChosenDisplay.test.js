import React from 'react';
import { shallow } from 'enzyme';
import { ChosenDisplay, mapStateToProps } from './ChosenDisplay';

describe('ChosenDisplay', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ChosenDisplay />);
  });

  it('should match the snapshot iwth all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {

  it('should return an object with the champions array', () => {
    const mockState = {
      champions: [{name: 'Varus'}]
    };
    const expected = {
      champions: [{name: 'Varus'}]
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});