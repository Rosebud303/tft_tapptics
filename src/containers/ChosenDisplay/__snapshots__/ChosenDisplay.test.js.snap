// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChosenDisplay should match the snapshot iwth all the data passed in 1`] = `undefined`;
